front/app/containers/Admin/projects/edit/general/utils/save.ts

```diff
@@ -1,108 +0,0 @@
-import { isEmpty, get, isString } from 'lodash-es';
-import eventEmitter from 'utils/eventEmitter';
-import { IParticipationContextConfig } from '../../participationContext';
-
-// i18n
-import messages from '../messages';
-
-// services
-import {
-  IUpdatedProjectProperties,
-  addProject,
-  updateProject,
-} from 'services/projects';
-import { addProjectFile, deleteProjectFile } from 'services/projectFiles';
-import { addProjectImage, deleteProjectImage } from 'services/projectImages';
-
-// typings
-import { INewProjectCreatedEvent } from '../../../all/CreateProject';
-
-export default async function save(
-  participationContextConfig: IParticipationContextConfig | null
-) {
-  if (this.validate() && !this.state.processing) {
-    const { formatMessage } = this.props.intl;
-    const {
-      project,
-      projectImages,
-      projectImagesToRemove,
-      projectFiles,
-      projectFilesToRemove,
-    } = this.state;
-    const projectAttributesDiff: IUpdatedProjectProperties = {
-      ...this.state.projectAttributesDiff,
-      ...participationContextConfig,
-    };
-
-    try {
-      this.setState({ saved: false });
-      this.processing$.next(true);
-
-      let isNewProject = false;
-      let projectId = project ? project.data.id : null;
-
-      if (!isEmpty(projectAttributesDiff)) {
-        if (project) {
-          await updateProject(project.data.id, projectAttributesDiff);
-        } else {
-          const project = await addProject(projectAttributesDiff);
-          projectId = project.data.id;
-          isNewProject = true;
-        }
-      }
```

Comment:
Refactored it a bit, but all in all very confusing to use the same function for a new and existing project.